src/pages/login.jsx

```diff
@@ -1,72 +1,69 @@
-import { SpvWalletClient} from "@bsv/spv-wallet-js-client";
-import {Box, Button, Container, TextField, Typography} from '@mui/material';
-import bsv from "bsv";
-import React, {useState} from 'react';
-import {SeverityPill} from "../components/severity-pill";
-import {useLocalStorage} from "../hooks/localstorage";
-import {useConfig} from "@4chain-ag/react-configuration";
-import {useModifyCredentials} from "../hooks/use-credentials";
-import logger from "../logger";
+import { SpvWalletClient } from '@bsv/spv-wallet-js-client';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import React, { useState } from 'react';
+import { SeverityPill } from '../components/severity-pill';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { useConfig } from '@4chain-ag/react-configuration';
+import { CredTypeAccessKey, CredTypeAdmin, CredTypeXPriv, useCredentials } from '../hooks/useCredentials';
+import logger from '../logger';
 
 const Login = () => {
-
   const { config } = useConfig();
   const [loginKey, setLoginKey] = useState('');
   const [serverUrl, setServerUrl] = useLocalStorage('login.serverUrl', config.serverUrl);
   const [error, setError] = useState('');
-  const {setAccessKeyString, setAdminKey, setServer, setXPrivString} = useModifyCredentials();
+  const { set: setCredentials } = useCredentials();
 
   const handleSubmit = async function (e) {
     e.preventDefault();
-    if (loginKey && (config.serverUrl || serverUrl)) {
-      let spvWalletClient;
-      let useServerUrl = serverUrl || config.serverUrl
-      try {
-        if (config.serverUrl) {
-          // use the hardcoded defaults for server url
-          useServerUrl = config.serverUrl;
-        }
+    if (!loginKey || !(config.serverUrl || serverUrl)) {
+      setError('Please set a server and an xPriv to connect');
+      logger.warn('Please set a server and an xPriv to connect');
+      return;
+    }
 
-        // try to make a connection and get the xpub
-        spvWalletClient = new SpvWalletClient(useServerUrl, {
-          xPrivString: loginKey.match(/^xprv/) ? loginKey : '',
-          accessKeyString: loginKey.match(/^[^xp]/) ? loginKey : '',
-          signRequest: true,
-        });
-        await spvWalletClient.GetXPub();
+    const currentServerUrl = config.serverUrl || serverUrl;
 
-        if (loginKey.match(/^xprv/)) {
-          bsv.HDPrivateKey.fromString(loginKey);
-          setXPrivString(loginKey);
-        } else {
-          bsv.PrivateKey.fromString(loginKey);
-          setAccessKeyString(loginKey);
-        }
-        setServer(useServerUrl);
+    // NOTE: the login procedure is as follows:
+    // 1. If the loginKey fits the xpriv pattern, then try to login with it:
+    // --- a. as xpriv
+    // --- b. if it fails, as admin key
+    // 2. If the login key does not fit the xpriv pattern, then try use loginKey as an access key
+
+    if (loginKey.match(/^xprv/)) {
```

Comment:
Check out this comment. Are we agree with that? (I've only refactorized the code)

Reply:
It's valid

Reply:
It seems right.

Reply:
So we don't allow a user to use the same xPriv both for normal methods and admin methods? This would be very useful for testing.